mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -83,14 +85,19 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	dep DepositT,
 ) error {
-	depositIndex, err := st.GetEth1DepositIndex()
-	if err != nil {
-		return err
+	var nextDepositIndex uint64
+	switch depositIndex, err := st.GetEth1DepositIndex(); {
+	case err == nil:
+		nextDepositIndex = depositIndex + 1
```

Comment:
nit for readability: add comment mentioning that no error means we can increment as expected

Reply:
Done